janusgraph-es/src/test/java/org/janusgraph/diskstorage/es/ElasticSearchIndexTest.java

```diff
@@ -254,9 +252,11 @@ public void testIngestPipeline() throws Exception {
         }
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void testMapKey2Field_IllegalCharacter() {
-        index.mapKey2Field("here is an illegal character: " + REPLACEMENT_CHAR, null);
+        assertThrows(IllegalArgumentException.class, () -> {
+            index.mapKey2Field("here is an illegal character: " + REPLACEMENT_CHAR, null);
+        });
```

Comment:
I haven't actually tested this, but I think if we want to capture that the error message is as expected we can update this to:
```
Throwable exception = assertThrows(IllegalArgumentException.class, () ->{
    index.mapKey2Field("here is an illegal character: " + REPLACEMENT_CHAR, null);
});
assertEquals("here is an illegal character: " + REPLACEMENT_CHAR, exception.getMessage());
```

Reply:
Same here. I don't think `"here is an illegal character: " + REPLACEMENT_CHAR`is the exception message.

Reply:
Agree here with @farodin91 also.